packages/sitecore-jss-angular/src/components/placeholder.component.spec.ts

```diff
@@ -844,3 +896,368 @@ describe('SXA components', () => {
     })
   );
 });
+
+describe('Placeholder Metadata:', () => {
+  const {
+    layoutData,
+    layoutDataWithEmptyPlaceholder,
+    layoutDataWithUnknownComponent,
+  } = metadataData;
+
+  @Component({
+    selector: 'test-text',
+    template: `
+      <div className="nested-test-wrapper"></div>
+    `,
+  })
+  class TestTextComponent {
+    @Input() rendering: ComponentRendering;
+  }
+
+  let fixture: ComponentFixture<TestPlaceholderComponent>;
+  let de: DebugElement;
+  let comp: TestPlaceholderComponent;
+
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        declarations: [TestTextComponent, TestPlaceholderComponent],
+        imports: [
+          RouterTestingModule,
+          JssModule.withComponents([{ name: 'Home', type: TestTextComponent }]),
+        ],
+        providers: [],
+      }).compileComponents();
+
+      fixture = TestBed.createComponent(TestPlaceholderComponent);
+      de = fixture.debugElement;
+
+      comp = fixture.componentInstance;
+      const jssContext = de.injector.get(JssStateService);
+      jssContext.setState(layoutData);
+      fixture.detectChanges();
+    })
+  );
+
+  it(
+    'should render code blocks around nested placeholder components with DEFAULT_PLACEHOLDER_UID',
```

Comment:
This unit test doesn't cover the case that we expected to cover. It doesn't render nested placeholder and doesn't have "assert" statement for it

Reply:
fixed.